### ISSUE TITLE
Interactive Shiny Quarto document are ran with rmarkdown instead of Quarto

### ISSUE THREAD

**Issue by cderv** [rstudio/rstudio#13207]
### System details

    RStudio Edition : Desktop [Open Source]
    RStudio Version : 2023.6.0.422
    OS Version      : Windows 10 x64 (build 22621)
    R Version       : R version 4.2.2 (2022-10-31 ucrt)

### Steps to reproduce the problem

* File > New File > Quarto Document > Interactive > Shiny Document
* Save the qmd file to `quarto-shiny.qmd`
* Click Run Document

### Describe the problem in detail

Render pane is trigger and calling `rmarkdown::run()` which errors because quarto should be called

````
==> rmarkdown::run('C:/Users/chris/Documents/quarto-shiny.qmd', shiny_args = list(launch.browser = FALSE), auto_reload = FALSE, dir = 'C:/Users/chris/Documents',  render_args = list(encoding = 'UTF-8', output_format = 'html'));
++ Activating rlang global_entrace

Error:
! objet 'html' introuvable
Backtrace:
    ▆
 1. └─rmarkdown::run(...)
 2.   └─rmarkdown:::shiny_prerendered_app(target_file, render_args = render_args)
 3.     └─rmarkdown:::shiny_prerendered_html(input_rmd, render_args)
 4.       ├─base::do.call(render, args)
 5.       └─rmarkdown (local) `<fn>`(...)
 6.         └─rmarkdown:::output_format_from_yaml_front_matter(...)
 7.           └─rmarkdown:::create_output_format_function(format_name)
Exécution arrêtée
````

### Describe the behavior you expected

`quarto serve` should be called with a `.qmd` file that has `format` key (and not `output:` key), and `server: shiny` key

https://quarto.org/docs/interactive/shiny/running.html

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by cderv** [comment]
Additional note: I am surprised this is an issue TBH. I believe this worked in the past but I did not test more versions. So maybe there was some regression at some point. Anyhow, you'll probably have a good process to confirm all this !

**Comment by jgutman** [comment]
@cderv This does not reproduce for me on Mac 13.6 (tested with RStudio 2023.06.2-561 and 2023.09.0-463) or on Windows Server 2019 (tested with RStudio 2023.06.3-575 and 2023.12.0-117). Renders successfully and does not call `rmarkdown::render`. Are you still seeing this problem?

**Comment by cderv** [comment]
Hi @jgutman

I cannot reproduce anymore indeed. Now I correctly get Quarto to render. Though the log message in Render pane is the following
````
==> (function(input, ...) { invisible(system(paste0('quarto serve "', input, '" ', ''))) })('C:/Users/chris/Documents/test-shiny.qmd', shiny_args = list(launch.browser = FALSE), auto_reload = FALSE, dir = 'C:/Users/chris/Documents',  render_args = list(encoding = 'UTF-8', output_format = 'html'));
````

As you see we have `invisible(system(paste0('quarto serve "', input, '" ', '')))` which shows quarto is ran. However, we have 
````
('C:/Users/chris/Documents/test-shiny.qmd', shiny_args = list(launch.browser = FALSE), auto_reload = FALSE, dir = 'C:/Users/chris/Documents',  render_args = list(encoding = 'UTF-8', output_format = 'html'));
````
and I am suspicious about this. It looks like part of the previous message was wrongly appended to the new log message when quarto is used. See previous message I got in OP
````
==> rmarkdown::run('C:/Users/chris/Documents/quarto-shiny.qmd', shiny_args = list(launch.browser = FALSE), auto_reload = FALSE, dir = 'C:/Users/chris/Documents',  render_args = list(encoding = 'UTF-8', output_format = 'html'));
++ Activating rlang global_entrace
````

I believe this is run for all Shiny doc (Rmd or .qmd)
https://github.com/rstudio/rstudio/blob/dbe70e7c6f6da18f2268d50e44c4e8002ba328ea/src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp#L570-L579
and used to create the command 
https://github.com/rstudio/rstudio/blob/dbe70e7c6f6da18f2268d50e44c4e8002ba328ea/src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp#L595-L601

This does not throw an issue because `...` is used in the `(function(input, ...) { invisible(system(paste0('quarto serve "', input, '" ', ''))) })`. However I think this should not be used with Quarto, and so it could be misleading to show it in log. 

Hope it helps

**Comment by jgutman** [comment]
I'll take a look at this again, but since it's not throwing an issue I'm going to deprioritize the milestone for now

**Comment by cderv** [comment]
> but since it's not throwing an issue I'm going to deprioritize the milestone for now

Yes it is just affecting log. You won't get any issue as `...` in R will eat the extra unused args. Only drawback of current situation is having some user's logging not accurant. that is all. 

I don't think you'll get any issue with it - you could consider this cosmetic I guess. 😅 Those type of things often stays deprioritize... 

Thanks again for checking this !

**Comment by ronblum** [comment]
I'd call it a step above cosmetic if we're giving incorrect information. 🙂 But I agree that it's a lower priority.

**Comment by cderv** [comment]
Understood. 👍 

Though, I guess we could consider it false as those args 
````
shiny_args = list(launch.browser = FALSE), 
auto_reload = FALSE, 
dir = 'C:/Users/chris/Documents',  
render_args = list(encoding = 'UTF-8', output_format = 'html')
````
have no usage at all when running `system(paste0('quarto serve "', input, '" ', ''))`  😄 

Anyhow, all good on this one as this is working fine now. Thanks !
